### PR TITLE
Fetch pool card data from contracts

### PR DESCRIPTION
**Pull request by lazarmitic**

### CODE REVIEW

File: packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts [hunk 1]
```diff
@@ -1056,4 +1065,111 @@ export default class PoolDataAdapter {
...
+    try {
+      const [tvlInBackingTokens, backingTokenRate] = await Promise.all([
+        this.statisticService.totalValueLockedInBackingTokens(tempusPool, {
+          blockTag: fetchForBlock,
+        }),
+        this.statisticService.getRate(backingToken, {
+          blockTag: fetchForBlock,
+        }),
+      ]);
+      const pastTVL = mul18f(tvlInBackingTokens, backingTokenRate);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts:1096]: we should pass `tokenPrecision` to `mul18f`

File: packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts [hunk 2]
```diff
@@ -1056,4 +1065,111 @@ export default class PoolDataAdapter {
...
+
+      const tvlDiff = currentTVL.sub(pastTVL);
+      const tvlRatio = div18f(tvlDiff, pastTVL);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts:1099]: same as above

File: packages/tempus-client_v2/src/components/pool/Pool.tsx [hunk 3]
```diff
@@ -1,14 +1,111 @@
...
+            activePoolData.backingTokenTicker,
+            activePoolData.tvl,
+          ),
+        );
+      } catch (error) {
+        console.error('fetchTVLChangeData() - Failed to fetch TVL change percentage!');
+      }
+    };
+    fetchTVLChangeData();
+  }, [activePoolData.address, activePoolData.backingTokenTicker, activePoolData.tvl, userWalletSigner]);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/components/pool/Pool.tsx:54]: I think we can pass `activePoolData` as dependency and simplify a bit

File: packages/tempus-client_v2/src/components/pool/Pool.tsx [hunk 4]
```diff
@@ -1,14 +1,111 @@
...
+            activePoolData.principalsAddress,
+          ),
+        );
+      } catch (error) {
+        console.error('fetchVolume() - Failed to fetch 7D volume for pool!');
+      }
+    };
+    fetchVolume();
+  }, [
+    activePoolData.address,
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/components/pool/Pool.tsx:81]: see my comment above

File: packages/tempus-client_v2/src/components/pool/Pool.tsx [hunk 5]
```diff
@@ -1,14 +1,111 @@
...
+    activePoolData.id,
+    activePoolData.principalsAddress,
+    userWalletSigner,
+  ]);
+
+  const tvlFormatted = useMemo(() => {
+    if (!activePoolData.tvl) {
+      return null;
+    }
+    return NumberUtils.formatWithMultiplier(ethers.utils.formatEther(activePoolData.tvl), 2);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/components/pool/Pool.tsx:92]: replace `formatEther` with `formatUnits` and pass token precision

File: packages/tempus-client_v2/src/components/pool/Pool.tsx [hunk 6]
```diff
@@ -1,14 +1,111 @@
...
+  }, [activePoolData.tvl]);
+
+  const tvlChangePercentageFormatted = useMemo(() => {
+    if (!tvlChangePercentage) {
+      return null;
+    }
+    return Number(ethers.utils.formatEther(tvlChangePercentage));
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/components/pool/Pool.tsx:99]: see my comment above

File: packages/tempus-client_v2/src/components/pool/Pool.tsx [hunk 7]
```diff
@@ -1,14 +1,111 @@
...
+  }, [tvlChangePercentage]);
+
+  const volumeFormatted = useMemo(() => {
+    if (!volume) {
+      return null;
+    }
+    return NumberUtils.formatWithMultiplier(ethers.utils.formatEther(volume), 2);
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/components/pool/Pool.tsx:106]: same as above

File: packages/tempus-client_v2/src/services/TempusControllerService.ts [hunk 8]
```diff
@@ -62,7 +62,12 @@ class TempusControllerService {
     this.tempusAMMService = params.tempusAMMService;
   }
 
-  public async getDepositedEvents(forPool?: string, forUser?: string): Promise<DepositedEvent[]> {
+  public async getDepositedEvents(
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/services/TempusControllerService.ts:65]: when having multiple optional arguments is better to pass an object like
`
{ forPool?: boolean, forUser?: boolean, fromBlock: number, toBlock?: number}
`

File: packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts (hunk 1 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts:1096]: Oh, right, forgot about that. Fixing it now.

File: packages/tempus-client_v2/src/services/TempusControllerService.ts [hunk 9]
```diff
@@ -71,14 +76,19 @@ class TempusControllerService {
...
       return Promise.reject(error);
     }
   }
 
-  public async getRedeemedEvents(forPool?: string, forUser?: string): Promise<RedeemedEvent[]> {
+  public async getRedeemedEvents(
+    forPool?: string,
+    forUser?: string,
+    fromBlock?: number,
+    toBlock?: number,
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/services/TempusControllerService.ts:90]: see my comment above

File: packages/tempus-client_v2/src/services/VaultService.ts [hunk 10]
```diff
@@ -68,7 +68,7 @@ class VaultService {
     this.tempusAMMService = params.tempusAMMService;
   }
 
-  public async getSwapEvents(forPoolId?: string, fromBlock?: number): Promise<SwapEvent[]> {
+  public async getSwapEvents(forPoolId?: string, fromBlock?: number, toBlock?: number): Promise<SwapEvent[]> {
```
Comment (danieledamiani) [inline packages/tempus-client_v2/src/services/VaultService.ts:71]: same comment as `TempusControllerService`

**Review by danieledamiani — CHANGES_REQUESTED** [review]
Some comments

File: packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts (hunk 1 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts]: fixed.

File: packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts (hunk 2 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/adapters/PoolDataAdapter.ts]: fixed.

File: packages/tempus-client_v2/src/components/pool/Pool.tsx (hunk 3 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/components/pool/Pool.tsx]: fixed.

File: packages/tempus-client_v2/src/components/pool/Pool.tsx (hunk 4 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/components/pool/Pool.tsx]: fixed.

File: packages/tempus-client_v2/src/components/pool/Pool.tsx (hunk 5 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/components/pool/Pool.tsx]: fixed.

File: packages/tempus-client_v2/src/components/pool/Pool.tsx (hunk 6 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/components/pool/Pool.tsx]: fixed.

File: packages/tempus-client_v2/src/components/pool/Pool.tsx (hunk 7 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/components/pool/Pool.tsx]: fixed.

File: packages/tempus-client_v2/src/services/TempusControllerService.ts (hunk 8 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/services/TempusControllerService.ts]: fixed.

File: packages/tempus-client_v2/src/services/TempusControllerService.ts (hunk 9 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/services/TempusControllerService.ts]: fixed.

File: packages/tempus-client_v2/src/services/VaultService.ts (hunk 10 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/services/VaultService.ts]: fixed.

File: packages/tempus-client_v2/src/components/pool/Pool.tsx (hunk 6 above)
Comment (lazarmitic) [reply packages/tempus-client_v2/src/components/pool/Pool.tsx]: fixed.

**Review by danieledamiani — APPROVED** [review]

**Pull request merged**